Failing macos tests - `RuntimeError: Not a datatype (not a datatype)`

macos-latest tests are failing since yesterday.

Probably related to yesterday's [h5py release](https://github.com/h5py/h5py/releases/tag/3.15.0), or a changed HDF5 homebrew package.

```
==================================== ERRORS ====================================
__________ ERROR collecting python/tests/test_pregenerated_models.py ___________
test_pregenerated_models.py:10: in <module>
    import h5py
../../venv/lib/python3.11/site-packages/h5py/__init__.py:45: in <module>
    from ._conv import register_converters as _register_converters, \
h5py/_conv.pyx:1: in init h5py._conv
    ???
h5py/h5r.pyx:1: in init h5py.h5r
    ???
h5py/h5p.pyx:1: in init h5py.h5p
    ???
h5py/h5t.pyx:152: in init h5py.h5t
    ???
h5py/h5t.pyx:86: in h5py.h5t.lockid
    ???
h5py/h5t.pyx:55: in h5py.h5t.typewrap
    ???
E   RuntimeError: Not a datatype (not a datatype)
__________ ERROR collecting python/tests/test_pregenerated_models.py ___________
test_pregenerated_models.py:10: in <module>
    import h5py
../../venv/lib/python3.11/site-packages/h5py/__init__.py:45: in <module>
    from ._conv import register_converters as _register_converters, \
h5py/_conv.pyx:1: in init h5py._conv
    ???
h5py/h5r.pyx:1: in init h5py.h5r
    ???
h5py/h5p.pyx:1: in init h5py.h5p
    ???
h5py/h5t.pyx:152: in init h5py.h5t
    ???
h5py/h5t.pyx:86: in h5py.h5t.lockid
    ???
h5py/h5t.pyx:55: in h5py.h5t.typewrap
    ???
E   RuntimeError: Not a datatype (not a datatype)
=========================== short test summary info ============================
ERROR test_pregenerated_models.py - RuntimeError: Not a datatype (not a datatype)
ERROR test_pregenerated_models.py - RuntimeError: Not a datatype (not a datatype)
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
============================== 2 errors in 0.37s ===============================
```

--- a/python/sdist/pyproject.toml
+++ b/python/sdist/pyproject.toml
@@ -63,7 +63,7 @@ classifiers = [
 petab = ["petab>=0.4.0"]
 pysb = ["pysb>=1.13.1"]
 test = [
-    "h5py!=3.15; platform_system=='Darwin'",
+    "h5py!=3.15",
     "pytest",
     "pytest-cov",
     # v16.0: https://github.com/pytest-dev/pytest-rerunfailures/issues/302

